### PR TITLE
sidevm: Support accessing local cache

### PR DESCRIPTION
**Pull request by kvinwang**

### CODE REVIEW

File: crates/pink/sidevm/host-runtime/src/env.rs [hunk 1]
```diff
@@ -83,6 +92,13 @@ impl TaskSet {
     }
 }
 
+pub struct CacheOps {
```
[the remark this answers is not in the document]
Comment (kvinwang) [reply crates/pink/sidevm/host-runtime/src/env.rs:95]: That would require trait bounds which either:
- Turn many types and functions into generic.
- or wrap the object in a box: `ops: Box<dyn CacheOps>` which would make it hard for the compiler to optimize the invocations.

Both of which I feel not good.

File: crates/pink/examples/start_sidevm/build.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -e
+
+cargo build --manifest-path sideprog/Cargo.toml --release --target wasm32-unknown-unknown
```
[the remark this answers is not in the document]
Comment (kvinwang) [reply crates/pink/examples/start_sidevm/build.sh:4]: Yeah, I personally prefer `make`. However, this script can explain the steps more clearly and can run without `make` installed.

File: crates/pink/sidevm/host-runtime/src/env.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kvinwang) [reply crates/pink/sidevm/host-runtime/src/env.rs:95]: Oh, there is a third option that the compiler might be able to optimize: `ops: &'static dyn CacheOps`.

[the remark this answers is not in the document]
Comment (kvinwang) [reply crates/pink/sidevm/host-runtime/src/env.rs]: Done the third approach.

File: crates/pink/examples/start_sidevm/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (h4x3rotab) [reply crates/pink/examples/start_sidevm/build.sh:4]: Ack

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**